src/test/java/io/confluent/examples/streams/CustomJoinWithTableTriggeringStreamTableJoinIntegrationTest.java

```diff
@@ -0,0 +1,429 @@
+   * not yet joined) record on the stream side.  If there is a match, then the transformer will produce a fully
+   * populated join output message -- which is the desired table-side triggering behavior.  If there is no match, then
+   * the transformer will do nothing.
+   */
+  private static final class StreamTableJoinTableSideTrigger
+    implements ValueTransformerWithKeySupplier<String, Long, Pair<Double, Long>> {
+
+    final private String streamBufferStoreName;
+
+    StreamTableJoinTableSideTrigger(final String streamBufferStoreName) {
+      this.streamBufferStoreName = streamBufferStoreName;
+    }
+
+    @Override
+    public ValueTransformerWithKey<String, Long, Pair<Double, Long>> get() {
+      return new ValueTransformerWithKey<String, Long, Pair<Double, Long>>() {
+
+        private KeyValueStore<String, Pair<Double, Instant>> streamBufferStore;
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public void init(final ProcessorContext context) {
+          streamBufferStore = (KeyValueStore<String, Pair<Double, Instant>>) context.getStateStore(streamBufferStoreName);
+        }
+
+        @Override
+        public Pair<Double, Long> transform(final String key, final Long value) {
+          return possiblySendFullJoinRecord(key, value);
+        }
+
+        private Pair<Double, Long> possiblySendFullJoinRecord(final String key, final Long value) {
+          if (value != null) {
+            final Pair<Double, Instant> streamValue = streamBufferStore.get(key);
+            if (streamValue != null) {
+              // We have data from both stream and table, so we can send a fully populated join message downstream.
+              streamBufferStore.delete(key);
+              return new Pair<>(streamValue.x, value);
+            } else {
+              return null;
+            }
+          } else {
+            return null;
+          }
+        }
+
+        @Override
+        public void close() {
+        }
+
+      };
+    }
+
+  }
+
+  @Test
+  public void shouldTriggerStreamTableJoinFromTable() throws Exception {
+
+    final List<KeyValue<String, Double>> inputStreamRecords = Arrays.asList(
+      new KeyValue<>("bob", 888.88),
+      new KeyValue<>("alice", 999.99)
+    );
+
+    final List<KeyValue<String, Long>> inputTableRecords = Arrays.asList(
+      new KeyValue<>("alice", 1L),
+      new KeyValue<>("alice", 2L)
+    );
+
+    final List<KeyValue<String, Pair<Double, Long>>> expectedRecords = Arrays.asList(
+      // The join output will have 2L (not 1L) for the table side.  That's because, by default, the DSL enables record
+      // caching for tables, which will cause the ValueTransformerWithKey for the KTable to not observe the 1L value.
+      //
+      // If your use case needs to produce a join output with value 1L, there are several options available.
+      //
+      // 1) Set `StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG` to `0`.  This will disable record caching,
```

Comment:
One can also disable caching for a single store via `Materialized.as(...).withCachingDisabled());`

Reply:
Ack, I added this to the updated example docs.